src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -102,6 +103,16 @@ internal abstract partial class AbstractLegacyProject : ForegroundThreadAffiniti
             // (e.g. through a <defaultnamespace> msbuild property)
             VisualStudioProject.DefaultNamespace = GetRootNamespacePropertyValue(hierarchy);
 
+            if (TryGetBoolPropertyValue(hierarchy, AdditionalPropertyNames.RunAnalyzers, out var runAnayzers))
+            {
+                VisualStudioProject.RunAnalyzers = runAnayzers;
+            }
```

Comment:
📝 I would not expect to read this property. I would expect to only read `RunAnalyzersDuringLiveAnalysis`, and have the project system do something like this:

```xml
<PropertyGroup>
  <RunAnalyzersDuringLiveAnalysis Condition="'$(RunAnalyzersDuringLiveAnalysis)' == ''">$(RunAnalyzers)</RunAnalyzersDuringLiveAnalysis>
</PropertyGroup>
```

Reply:
The implementation in this PR matches the current implementation in the core code analysis targets file, project system and our documentation. Feel free to file a separate issue if you want to propose a different approach/set of properties.

1. `Microsoft.CodeAnalysis.targets` implementation:
```
  <PropertyGroup>
    <IsCSharpOrVbProject Condition="'$(Language)'=='C#' or '$(Language)'=='VB'">true</IsCSharpOrVbProject>
    <RunAnalyzersDuringLiveAnalysis Condition="'$(RunAnalyzersDuringLiveAnalysis)' == ''">$(IsCSharpOrVbProject)</RunAnalyzersDuringLiveAnalysis>
    <RunAnalyzersDuringBuild Condition="'$(RunAnalyzersDuringBuild)' == ''">$(IsCSharpOrVbProject)</RunAnalyzersDuringBuild>
  </PropertyGroup>

  <Target Name="ComputeRunAnalyzers"
           BeforeTargets="CoreCompile"
           Condition="'$(RunAnalyzers)' == ''">
    <PropertyGroup>
      <IsAnyDesignTimeBuild Condition="'$(DesignTimeBuild)' == 'true' or '$(BuildingProject)' != 'true'">true</IsAnyDesignTimeBuild>
      <RunAnalyzers Condition="'$(IsAnyDesignTimeBuild)' == 'true' and '$(RunAnalyzersDuringLiveAnalysis)' == 'false'">false</RunAnalyzers>
      <RunAnalyzers Condition="'$(IsAnyDesignTimeBuild)' != 'true' and '$(RunAnalyzersDuringBuild)' == 'false'">false</RunAnalyzers>
    </PropertyGroup>
  </Target>
  
  <Target Name="DisableAnalyzers"
           AfterTargets="ComputeRunAnalyzers"
           Condition="'$(RunAnalyzers)' == 'false'">
    <!-- 
       Disable analyzers via an MSBuild property settable on the command line. 
     -->
    <ItemGroup>
      <Analyzer Remove="@(Analyzer)" />
    </ItemGroup>
  </Target>
```

2. Documentation: https://docs.microsoft.com/en-us/visualstudio/code-quality/disable-code-analysis?view=vs-2019

Reply:
Is `RunAnalyzers` even set when you load the project in Visual Studio? I would expect it to not be set since the `ComputeRunAnalyzers` target hasn't run.

Reply:
Not from the target, and that is the reason why I have to read both the property values to determine the correct state. Maybe we can take the simpler route and avoid a separate `RunAnalyzers` property all together. The `DisableAnalyzers` target can be conditioned on `IsAnyDesignTimeBuild`, `RunAnalyzersDuringLiveAnalysis` and `RunAnalyzersDuringBuild`